src/main/java/htsjdk/samtools/filter/TagFilter.java

```diff
@@ -81,7 +108,6 @@ public boolean filterOut(SAMRecord record) {
      */
     @Override
     public boolean filterOut(final SAMRecord first, final SAMRecord second) {
-        // both first and second must have the tag in order for it to be filtered out
-         return values.contains(first.getAttribute(tag)) && values.contains(second.getAttribute(tag));
+        return (values.contains(first.getAttribute(tag)) && values.contains(second.getAttribute(tag))) != includeReads;

```

Comment:
hmmm. I'm actually not sure that this does the right thing...

if includeReads==false, and one has the tag/value and the other doesn't...should the pair be filtered or not?

at any rate...this should be tested. I know it wasn't already tested...but it should. could you please?

Reply:
👍

Reply:
I think for now I will assume the tag value has to be on both (or vice versa not on both) in order to be filtered/allowed because thats how it originally was implemented.  Is there a case for this not to be the logic?

Reply:
actually can we talk about this when you have some time